src/video/wayland/SDL_waylandsym.h

```diff
@@ -84,9 +84,11 @@ SDL_WAYLAND_MODULE(WAYLAND_CLIENT_1_18)
 SDL_WAYLAND_SYM(void, wl_proxy_set_tag, (struct wl_proxy *, const char * const *))
 SDL_WAYLAND_SYM(const char * const *, wl_proxy_get_tag, (struct wl_proxy *))
 
+#if defined(SDL_VIDEO_DRIVER_WAYLAND_DYNAMIC) || SDL_WAYLAND_CHECK_VERSION(1, 20, 0)
 SDL_WAYLAND_MODULE(WAYLAND_CLIENT_1_20)
```

Comment:
In SDL terminology, "shared" is ordinary ELF `DT_NEEDED` linking (`-lwayland-client` at link time), and "dynamic" is `dlopen()`, right? (I'm often confused by this, because both are dynamically linking a shared library - but I can't think of a less ambiguous word!)

Is what you're proposing this, so that the 1.20 symbols are sort of treated as though they were part of 1.18?

```
#if SDL_WAYLAND_CHECK_VERSION(1, 20, 0)
SDL_WAYLAND_SYM(struct wl_proxy*, wl_proxy_marshal_flags, (struct wl_proxy *proxy, uint32_t opcode, const struct wl_interface *interfac, uint32_t version, uint32_t flags, ...))
SDL_WAYLAND_SYM(struct wl_proxy*, wl_proxy_marshal_array_flags, (struct wl_proxy *proxy, uint32_t opcode, const struct wl_interface *interface, uint32_t version,  uint32_t flags, union wl_argument *args))
#endif
```

If so, then yes I think that makes sense.

> but at the same time, maybe don't use the shared path unless you want to be linking to both X and Wayland objects at the same time...

Well, yes, that's what most other GUI libraries that support Wayland end up doing (GTK, Clutter, KDE libraries, and so on). For portable binaries intended to be moved between runtime environments, yes it's undesirable, but for integrated OS distributions whose package managers have dependency resolution, the advantages of `DT_NEEDED`-based linking often outweigh the disadvantages.

Reply:
Yeah, my bad on the terminology - in SDL's world, "shared" means "dynamic link" and "dynamic" means "dynamic load".

But yeah, that proposal is what I'm thinking. If that can be applied in a rebase I can merge the full patch in!